src/components/LayoutTree.tsx

```diff
@@ -364,16 +370,24 @@ const LayoutTree = () => {
         state.cursor.length - (hasChildren(state, head(state.cursor)) ? 2 : 3)
       : 0,
   )
+  const scrollTop = scrollTopStore.useState()
 
   const [bulletWidth, setBulletWidth] = useState<number | undefined>(0)
+  const toolbarRef = useRef<number | undefined>(0)
+  const distanceFromTop = useRef<number | undefined>(0)
 
   // set the bullet width only during drag or when simulateDrop is true
   useLayoutEffect(() => {
     if (dragInProgress || testFlags.simulateDrop) {
       const bullet = ref.current?.querySelector('[aria-label=bullet]')
+      const toolbar = document.querySelector('#toolbar')
+
+      distanceFromTop.current = (ref.current?.getBoundingClientRect().top ?? 0) + scrollTop
+
+      toolbarRef.current = toolbar?.getBoundingClientRect().height
```

Comment:
Well, yes I avoided using state, since for most of the time `toolbar` and `distanceFropTop` will not be changing unless we change the font size. So, keeping it inside ref and only updating it when `dragInProgress` is true.

Reply:
If we use `useState`, it shouldn't trigger a rerender if the value hasn't changed (React does a shallow comparison), so there shouldn't be any unusual performance hit. I still believe it would make more sense for this to be a state value rather than a ref. Please let me know if I'm missing something.